client/src/game/ui/globals.ts

```diff
@@ -101,7 +101,7 @@ export class Globals {
   useSharedTurns: boolean = false;
   sharedReplayLoading: boolean = false; // This is used to not animate cards when loading in
   hypothetical: boolean = false; // Whether or not we are in a hypothetical
-  hypoActions: Action[] = []; // An array of the actions in the current hypothetical
+  hypoActions: ActionIncludingHypothetical[] = []; // Actions in the current hypothetical
```

Comment:
This will fix the fact that the reveal action was not being processed correctly sometimes